ca_description/urdf/create_light_sensor.xacro

```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0"?>
+<robot xmlns:xacro="http://ros.org/wiki/xacro">
+
+<xacro:macro name="light_sensor" params="name_prefix sensor_color *origin">
+
+    <xacro:include filename="$(find ca_description)/urdf/common_properties.xacro"/>
+    
+    <xacro:property name="name" value="${name_prefix}_light_sensor"/>
+    <xacro:property name="parent_link" value="base_link"/>
+
+        <joint name="${name}_joint" type="fixed">
+            <xacro:insert_block name="origin"/>
+            <parent link="${parent_link}" />
+            <child link="${name}_link" />
+        </joint>
+
+        <link name="${name}_link">
+        <visual>
+            <geometry>
+                <mesh filename="package://ca_description/meshes/last_lens.stl" />
+            </geometry>
+            <material name="blue">
+                <color rgba="0 0 1.0 1.0"/>
+            </material>
+        </visual>
+        </link>
+
+        <joint name="${name}_optical_joint" type="fixed">
+            <origin xyz="0.052 0 0" rpy="${-pi/2} 0 ${-pi/2}"/>/>
+            <parent link="${name}_link" />
+            <child link="${name}_optical_frame"/>
+        </joint>
+        
+        <link name="${name}_optical_frame"/>
+
+        <gazebo reference="${name}_link">
+            <material>Gazebo/DarkGrey</material>
+        </gazebo>
+
+        <gazebo reference="${name}_link">
+            <sensor name='camera' type='camera'>
+            <camera name='__default__'>
+                <horizontal_fov>0.5</horizontal_fov>
+                <image>
+                <width>40</width>
```

Comment:
Can't be smaller?

Reply:
I tried with smaller values, and it's not working so well. I'll keep it at 40. At least for now.